### PR TITLE
Implement generator method.

### PR DESCRIPTION
**Pull request by samuelwilliams**
Resolves Issue #23

### CODE REVIEW

File: tests/IPBlockTest.php [hunk 1]
```diff
@@ -209,4 +211,36 @@ public function testGetSuper()
         } catch (\InvalidArgumentException $e) {
         }
     }
+
+    /**
+     * Tests the generator
+     */
+    public function testGetAddresses()
+    {
```
Comment (rlanvin) [inline tests/IPBlockTest.php]: I do not like these 2 tests, they test too many different hings (relies on dynamic computation, humanReadable, one test ipv4 and the other one ipv6, etc.). I would prefer to have a data provider that provides a block and an preset array of expected IPs. Then the test just run the generator and collects all the IP into an array, and assert that the collected array equals the expected array. 16 IPs is still reasonable to write in a data provider, but you can also use smaller blocks. The 2 tests can use the same data provider obviously. What do you think?

Comment (samuelwilliams) [reply tests/IPBlockTest.php]: That seems like a good approach. I'll amend the PR.

**Comment by samuelwilliams** [comment]
So I've made different tests, but upon reflection, should we even have a method called `getAddresses()` in addition to `getIterator()`? I think a single method (`getIterator`) is sufficient. Thoughts?

**Comment by rlanvin** [comment]
@samuelwilliams I agree, that was also my thought, I usually implement the generator directly in getIterator when I use IteratorAggregate.

**Comment by samuelwilliams** [comment]
I've deleted the `getAddresses()` method.

**Comment by rlanvin** [comment]
Thanks!

**Pull request merged**